lib/emscripten/src/lib.rs

```diff
@@ -384,8 +410,12 @@ pub fn run_emscripten_instance(
             .get(&ep)
             .map_err(|e| RuntimeError::new(e.to_string()))?;
         func.call(&[Val::I32(arg as i32)])?;
+    } else if let Ok(name) = emscripten_get_main_func_name(instance, &main_func_names) {
```

Comment:
Ignoring the error case from `emscripten_get_main_func_name`; maybe it should return an Option then instead?

Reply:
Isn't a Result preferable for a public library function?

Reply:
I'd stick with `Result` because `emscripten_call_main` also has a result. But I would document the function better.